hypothesis-python/src/hypothesis/searchstrategy/regex.py

```diff
@@ -182,7 +182,11 @@ def add_char(self, char):
         """Add given char to the whitelist."""
         c = self.code_to_char(char)
         self._whitelist_chars.add(c)
-        if self._ignorecase and re.match(c, c.swapcase(), re.IGNORECASE) is not None:
+        if (
+            self._ignorecase
+            and c != c.swapcase
+            and re.match(c, c.swapcase(), re.IGNORECASE) is not None
```

Comment:
This still seems not great in that it works basically by coincidence - you're relying on the fact that `c != swapcase` implies that `c` is a valid regex. This happens to be true (I think) but doesn't really communicate why it works at all.

Given that there is a method that is explicitly designed to turn `c` into a valid regex that would much more clearly communicate the intent, I'm still not clear on why you're not just using that.